pulse2percept/models/granley2021.py

```diff
@@ -440,7 +451,17 @@ def _predict_spatial(self, earray, stim):
         x = np.array(x, dtype=np.float32)
         y = np.array(y, dtype=np.float32)
 
-        if self.engine != 'jax':
+
+        if self.engine == 'torch':
+            x_el = np.array([earray[e].x for e in stim.electrodes], dtype=np.float32)
+            y_el = np.array([earray[e].y for e in stim.electrodes], dtype=np.float32)
+            e_locs = torch.tensor([(x,y) for x,y in zip(x_el, y_el)]).to(self.device)
```

Comment:
Please prefer to use tensor(..., device=device) instead of tensor(...).to(device) to avoid unnecessary copy